hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/config/MetricsExecutionInterceptorTest.java

```diff
@@ -0,0 +1,216 @@
+
+        var prefix =
+                switch (pathType) {
+                    case ACCOUNT_ID, AUTO -> accountIdPrefix(streamType, SHARD, REALM, ACCOUNT_NUM);
+                    case NODE_ID -> nodeIdPrefix(streamType, NETWORK_NAME, SHARD, NODE_ID);
+                };
+
+        var objectKey = prefix + fileName;
+        var sdkHttpRequest = createGetObjectRequest(objectKey);
+
+        when(afterExecutionContext.httpResponse()).thenReturn(sdkHttpResponse);
+        when(sdkHttpResponse.statusCode()).thenReturn(HTTP_STATUS_SUCCESS);
+        when(afterExecutionContext.httpRequest()).thenReturn(sdkHttpRequest);
+
+        metricsExecutionInterceptor.afterExecution(afterExecutionContext, executionAttributes);
+        verifyTimerTags(expectedAction, NODE_ID, streamType);
+    }
+
+    /*
+     * Test failure scenario where the SDK uri does not have a parsable bucket path. The resultant inner
+     * IllegalStateException is caught and logged and there is no visibility of that. However, a Timer
+     * will not have been created, so verify that is the case.
+     */
+    @Test
+    void invalids3GetObjectExecution() {
+        var sdkHttpRequest = createGetObjectRequest("uripaththatdoesnotmatchregex");
+        when(afterExecutionContext.httpRequest()).thenReturn(sdkHttpRequest);
+
+        metricsExecutionInterceptor.afterExecution(afterExecutionContext, executionAttributes);
+        Collection<Timer> timers =
+                meterRegistry.find("hedera.mirror.download.request").timers();
+        assertEquals(0, timers.size());
+    }
+
+    private String nodeIdPrefix(StreamType streamType, String network, long shard, long nodeId) {
+        return "%s/%d/%d/%s/".formatted(network, shard, nodeId, streamType.getNodeIdBasedSuffix());
+    }
+
+    private String accountIdPrefix(StreamType streamType, long shard, long realm, long accountNum) {
+        return "%s/%s%d.%d.%d/".formatted(streamType.getPath(), streamType.getNodePrefix(), shard, realm, accountNum);
+    }
+
+    private void verifyTimerTags(String expectedAction, long expectedNodeId, StreamType expectedStreamType) {
+        Collection<Timer> timers =
+                meterRegistry.find("hedera.mirror.download.request").timers();
+        assertEquals(1, timers.size());
+
+        List<Tag> providedTags = timers.iterator().next().getId().getTags();
+        assertNotNull(providedTags);
+        assertEquals(6, providedTags.size());
+
+        // The tags are in a defined order, with the tag name followed by its value. All are Strings.
+        assertEquals(expectedAction, providedTags.get(0).getValue());
+        assertEquals("GET", providedTags.get(1).getValue());
+        assertEquals(expectedNodeId, Long.valueOf(providedTags.get(2).getValue()));
+        assertEquals("0", providedTags.get(3).getValue());
+        assertEquals("200", providedTags.get(4).getValue());
+        assertEquals(expectedStreamType.name(), providedTags.get(5).getValue());
```

Comment:
We prefer using assertThat() over JUnit asserts and chaining multiple together.

```suggestion
        assertThat(meterRegistry.find("hedera.mirror.download.request").timers())
            .hasSize(1)
            .first()
            .returns(1L, t -> t.count())
            .returns(?, t -> t.mean(TimeUnit.SECONDS))
            .returns(?, t -> t.max(TimeUnit.SECONDS))
            .returns(expectedAction, t -> t.getId().getTag(TAG_ACTION))
            ...
```